### ISSUE TITLE
Should it work fine if I'm using HashRouter?

### ISSUE THREAD

**Issue by vcardins** [Dean177/with-url-state#175]

**Comment by Dean177** [comment]
Yes, the router you use shouldn't have any impact on functionality/

Are you encountering unexpected behaviour

**Comment by vcardins** [comment]
Hi Dean, thanks for the quick reply. Yes, I noticed that it removes the hash route:

I decorated my class component like
`const MyPagePageWithUrlState = withUrlState(() => Utils.getSearchParams())(MyPage);`

then when I click on a link to that page
`http://localhost/#/page?&filter1=2190&filter2=2&filter3=q4...`
it loads the page but replaces the URL to
`http://localhost/?&filter1=2190&filter2=2&filter3=q4#/page` (not the har route it was to the end)

Any advice?

**Comment by Dean177** [comment]
Are you using the v1.0.0-alpha.0? If not, perhaps give that a shot first.

If the newer version isn't working either you can pass a `history` object as a config parameter, it needs to have the following signature:
```
type UnregisterCallback = () => void

export type HistoryAdapter = {
  listen: (listener: () => void) => UnregisterCallback
  location: Location
  push: (location: Location) => void
  replace: (location: Location) => void
}

// Maybe your hash router has a compatible API that would let you use it like so:
let hashHistory = {
  listen: (listener) => hashRouter.listen(listener)
  location: hasRouter.location,
  push: (location) => hashRouter.push(location)
  replace: (location) => hashRouter.replace(location)  
}
```

What are you using for routing?

**Comment by vcardins** [comment]
Thanks, Dean! I'll try that. I'm using React Router 5.1.2.
